packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -45,13 +45,20 @@ const listShadowableFilesForTheme = (directory, theme) => {
 }
 
 const getOptionsForPlugin = node => {
-  if (!t.isObjectExpression(node)) {
+  if (!t.isObjectExpression(node) && !t.isLogicalExpression(node)) {
     return undefined
   }
 
-  const options = node.properties.find(
-    property => property.key.name === `options`
-  )
+  let options
+
+  // When a plugin is added conditionally with && {}
```

Comment:
It's not a blocker but I wanted to say that this opens you up to Pandora's box:

```js
// Conditionally existing config objects almost requires you to follow both paths
// and even then you can't statically determine which ends up being used, anyways
configa || configb;
// Ternary (which is a ConditionalExpression)
staging ? stagingConfig : prodConfig
```

One way of preventing the static determination is to say this sort of config is only supported with a strict set of env vars, checked in a very explicit way

```
process.env.STAGING ? {..} : {..}
```

This way you can be reasonably safe to assume what `process.env.XX` means, sort of like a pseudo-DSL. And power users can still do most of what they want to do.

It's a bit of feature creep and I wonder how many paper cuts the current way will introduce down the line.

In the end users might not expect the config to be statically analyzed this way. Assuming "it's just JS I can do whatever", leading to the exact kind of problem this PR is trying to solve. Over and over again.

Having some kind of explicit warning about unexpected structures in the config might help here, pointing to a doc explaining what's up. But that's a bit more work to do.

Reply:
We've discussed this in Slack, too, and the consensus was that apart from those changes it's not planned to support more cases -- as you said, it's a pandora's box. If ever change our mind on this we need to rethink this for sure instead of tacking on things.